### ISSUE TITLE
performance issues

### ISSUE THREAD

**Issue by utoni** [go-gitea/gitea#7176]
- Gitea version (or commit ref): 835b53fc259c82f38945a3e107a4eb51478967d5
- Git version: 2.11.0 && 2.21.0
- Operating system: Arch && Debian && OpenWrt
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes, see https://try.gitea.io/test1212et/kubernetes
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
I'm using gitea for quite a while now for my private projects.
Performance was always an issue for me. I've written a small patch to
increase the performance for large repositories. Basically by limiting the
max commit count for the git command with `--max-count`. The performance
was still not perfect but much better than before.

But after 1.8.1 it all got worse. The performance not only for large repo's
but additionally for smaller ones (<1000 commits) is pretty awful.
It takes 1 minute and more to load a large repo on a modern i9 and several SSD's in raid.
I verified that this issue is not related to the git executable.
It seems that it is related to the go code. I'm still working on that to figure
out what causes those heavy load times.

Any help would be appreciated.

**Comment by lunny** [comment]
Please change your app.ini to configure `[cache]` section. Counting commits will spend most time, and you could enable cache.

**Comment by techknowlogick** [comment]
Also upgrading to 1.9.0-dev would help speed things up too, as major refactoring has occurred.

**Comment by MarkusAmshove** [comment]
@lunny the config says the defaults are memory with TTL 16 hours.
If there is no `[cache]` setting in my config, is it still configured with the default values?

**Comment by lnslbrty** [comment]
> Please change your app.ini to configure `[cache]` section. Counting commits will spend most time, and you could enable cache.

I played around with the `[cache]` options and it literally changed ... nothing. Same initial loading times. Same loading times after page refresh. What does the "cache" actually cache?
It seems to me that this is a SnakeOil feature.

> Also upgrading to 1.9.0-dev would help speed things up too, as major refactoring has occurred.

I wouldn't write that issue if I did not try that first.
Yeah, I saw some of the spaghetti refactored code. But readability and performance did not change.
